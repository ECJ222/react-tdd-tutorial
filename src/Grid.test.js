/* eslint-disable no-undef */
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Grids from './components/gridcontainer'

afterEach(cleanup)

describe('Props test', () => {
  test('Should display exact text in all photo cards.', () => {
    const { getAllByText } = render(<Grids text="New text" />)
    const elements = getAllByText('New text')

    elements.forEach((element) => {
      expect(element.textContent).toBe('New text')
    })
  })
})

describe('Snapshot test', () => {
  test('Matches snapshot.', () => {
    const { getAllByText } = render(<Grids text="New text" />)
    const elements = getAllByText('New text')

    expect(elements).toMatchSnapshot()
  })
})
