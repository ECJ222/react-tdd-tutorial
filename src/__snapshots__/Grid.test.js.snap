// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Matches snapshot. 1`] = `
Array [
  <em
    class="sc-eCssSg fUldDc"
  >
    New text
  </em>,
  <em
    class="sc-eCssSg fUldDc"
  >
    New text
  </em>,
  <em
    class="sc-eCssSg fUldDc"
  >
    New text
  </em>,
  <em
    class="sc-eCssSg fUldDc"
  >
    New text
  </em>,
  <em
    class="sc-eCssSg fUldDc"
  >
    New text
  </em>,
  <em
    class="sc-eCssSg fUldDc"
  >
    New text
  </em>,
]
`;
